packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -19,40 +19,23 @@ function buildConstructor(classRef, constructorBody, node) {
   return func;
 }
 
-const verifyConstructorVisitor = traverse.visitors.merge([
-  environmentVisitor,
-  {
-    Super(path, state) {
-      if (state.isDerived) return;
-
-      const { node, parentPath } = path;
-      if (parentPath.isCallExpression({ callee: node })) {
-        throw path.buildCodeFrameError(
-          "super() is only allowed in a derived constructor",
-        );
-      }
-    },
-
-    ThisExpression(path, state) {
-      if (!state.isDerived) return;
-
-      const { node, parentPath } = path;
-      if (parentPath.isMemberExpression({ object: node })) {
-        // In cases like this.foo or this[foo], there is no need to add
-        // assertThisInitialized, since they already throw if this is
-        // undefined.
-        return;
-      }
-
-      const assertion = t.callExpression(
-        state.file.addHelper("assertThisInitialized"),
-        [node],
-      );
-      path.replaceWith(assertion);
-      path.skip();
+const verifyConstructorVisitor = (path, visitors) =>
```

Comment:
What is the `path` argument needed for?

Reply:
y, no used any more, i've put the visitor as a const.